x-pack/plugins/security_solution/public/management/components/endpoint_responder/command_render_components/kube_list_action.tsx

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import styled from 'styled-components';
+import moment from 'moment';
+import _ from 'lodash';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useConsoleActionSubmitter } from '../hooks/use_console_action_submitter';
+import type {
+  KubeListActionOutputContent,
+  KubeRequestBody,
+} from '../../../../../common/endpoint/types';
+import { useSendGetKubeListRequest } from '../../../hooks/response_actions/use_send_get_kube_list_request';
+import type { ActionRequestComponentProps } from '../types';
+
+const startTimeToAge = (startTime: string) => {
+  const diffInDays = moment(moment.now()).diff(moment(startTime), 'days');
+  const diffInHours = moment(moment.now()).diff(moment(startTime), 'hours');
+  const remainingHours = diffInHours - diffInDays * 24;
+
+  if (diffInDays > 0) {
+    return `${diffInDays}d${remainingHours > 0 ? `${remainingHours}h` : ''}`;
+  }
+  const diffInMinutes = moment(moment.now()).diff(moment(startTime), 'minutes');
+  const remainingMinutes = diffInMinutes - diffInHours * 60;
+
+  if (diffInHours > 0) {
+    return `${diffInHours}h${remainingMinutes > 0 ? `${remainingMinutes}m` : ''}`;
+  }
+
+  const diffInSeconds = moment(moment.now()).diff(moment(startTime), 'seconds');
+  const remainingSeconds = diffInSeconds - diffInMinutes * 60;
+
+  if (diffInMinutes > 0) {
+    return `${diffInMinutes}m${remainingSeconds > 0 ? `${remainingSeconds}s` : ''}`;
+  }
+
+  return `${diffInSeconds}s`;
+};
+
+// @ts-expect-error TS2769
+const StyledEuiBasicTable = styled(EuiBasicTable)`
+  table {
+    background-color: ${({ theme: { eui } }) => eui.euiPageBackgroundColor};
+  }
+  .euiTableHeaderCell {
+    border-bottom: ${(props) => props.theme.eui.euiBorderThin};
+    .euiTableCellContent__text {
+      font-weight: ${(props) => props.theme.eui.euiFontWeightRegular};
+    }
+  }
+  .euiTableRow {
+    &:hover {
+      background-color: white !important;
+    }
+    .euiTableRowCell {
+      border-top: none !important;
+      border-bottom: none !important;
+    }
+  }
+`;
+
+const COLUMNS_TO_RESOURCE_MAPPING = {
```

Comment:
can you use `deepFreeze` on this data structure in order to avoid accidental data mutations? or maybe convert it to a getter function instead, which will ensure a new object is returned every time.